task_processing/interfaces/event.py

```diff
@@ -23,17 +26,19 @@ class Event(PRecord):
 
 
 def json_serializer(o):
-    if isinstance(o, datetime.datetime):
-        return o.strftime('%s')
+    if isinstance(o, uuid.UUID):
+        print("encoded value of o %s is %s" % (o, o.hex))
+        return o.hex
+    return json.JSONEncoder.default(o)
 
 
 def json_deserializer(dct):
     for k, v in dct.items():
-        if k == 'timestamp':
+        if k == "uuid":
```

Comment:
because we still do a 'freeze' on the other vals to turn them into pyrsistent types